sdk/program/src/account_info.rs

```diff
@@ -30,12 +30,19 @@ pub struct AccountInfo<'a> {
 
 impl<'a> fmt::Debug for AccountInfo<'a> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use fmt::Write;
+
         let data_len = cmp::min(64, self.data_len());
         let data_str = if data_len > 0 {
-            format!(
-                " data: {} ...",
-                hex::encode(self.data.borrow()[..data_len].to_vec())
-            )
+            let mut s = " data: ".to_owned();
+
+            for &byte in &self.data.borrow()[..data_len] {
+                write!(s, "{:02x}", byte)?;
+            }
+
+            s.push_str(" ...");
```

Comment:
What do you think about a helper function to dedupe the to fns you touched?
Something like:
```
debug_fmt_account_data(data: &[u8], f: &mut fmt::Formatter<'_>) -> fmt::Result {
    const MAX_DATA_LEN: usize = 64;
    let data_len = cmp::min(64, self.data_len());
    let data_str = if data_len > 0 {
        write!(f, " data: ")?;
        for &byte in &data[..data_len] {
            write!(f, "{:02x}", byte)?;
        }
    }
}
```

I think it is not critical whether `Debug for AccountInfo` drops the ellipsis, or `debug_fmt` adds it, but I will double check.

Reply:
There is one usage in `sdk/` and another in `sdk/program`, and I don't think you can reuse code across the two.

Reply:
I don't see why not. solana-program is a dependency of solana-sdk.